Microsoft.Toolkit.Uwp.UI.Controls/GridSplitter/GridSplitter.Helper.cs

```diff
@@ -30,41 +30,105 @@ private static bool IsStarRow(RowDefinition definition)
             return ((GridLength)definition.GetValue(RowDefinition.HeightProperty)).IsStar;
         }
 
-        private void SetColumnWidth(ColumnDefinition columnDefinition, double horizontalChange, GridUnitType unitType)
+        private bool SetColumnWidth(ColumnDefinition columnDefinition, double horizontalChange, GridUnitType unitType)
         {
             var newWidth = columnDefinition.ActualWidth + horizontalChange;
+
+            var minWidth = columnDefinition.MinWidth;
+            if (minWidth != double.NaN && newWidth < minWidth)
+            {
+                newWidth = minWidth;
+            }
+
+            var maxWidth = columnDefinition.MaxWidth;
+            if (maxWidth != double.NaN && newWidth > maxWidth)
+            {
+                newWidth = maxWidth;
+            }
+
             if (newWidth > ActualWidth)
             {
-                columnDefinition.Width = new GridLength(newWidth, unitType);
+                if (columnDefinition.Width.Value != newWidth)
```

Comment:
This condition 99.9% of time is true, it is just adding overhead in the calculation as this method is executed multiple times with every manipulation.